### PR TITLE
V9.0

### PR DESCRIPTION
**Pull request by lukasbelak**
Hi @scottdurow,
As I promised, I push the webresource dependency support changes. Hopefully I am doing it right, I am a GitHub newbie. 

Here is an example of spkl.json:

{
	"webresources": [
		{
			"profile": "default,debug",
			"root": "webresources/new_/",
			"solution": "SpklTest",
			"files": [
				{
					"uniquename": "new_/js/account.js",
					"displayname": "new_/js/account.js",
					"file": "js\\account.js",
					"description": "desc",
					"dependencies": {
						"libraries": [
							{
								"uniquename": "new_/js/dg.xrmquery.web.min.js",
								"file": "js\\dg.xrmquery.web.min.js",
								"description": "desc",
								"displayname": "new_/js/dg.xrmquery.web.min.js"
							}
						],
						"attributes": [
							{
								"attributename": "accountid",
								"entityname": "account"
							}
						]
					}
				}
			]
		}
	]
}

I think the json itself it is self explanatory, but if you will have questions, do not hesitate to contact me, please.

Basically I made changes to 'deploy' and to 'download' webresources functionality. 
Regarding deploy, I only add dependency libraries that are store in CRM, I do not compare what is included in spkl.json.
Regarding download, I do not updatae existing files in spkl.json, if there is for example new dependency library added manually in CRM for a particular webresource. This could be also added, it is your decision.

Please, have a look at the code and you can also share your comments, concerns, whatever with me. I also included few comments.
Thanks.

### CODE REVIEW

**Comment by scottdurow** [comment]
Can you merge this into the spkl_dev branch - then I'll get it pulled in and create a new beta release.

**Comment by e-gauthier** [comment]
Hello all, 

Just want to ask what happend to this branch?  Is it still in dev?

Thx
